config/initializers/devise.rb

```diff
@@ -0,0 +1,313 @@
+# frozen_string_literal: true
+
+# Assuming you have not yet modified this file, each configuration option below
+# is set to its default value. Note that some are commented out while others
+# are not: uncommented lines are intended to protect your configuration from
+# breaking changes in upgrades (i.e., in the event that future versions of
+# Devise change the default values for those options).
+#
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  # The secret key used by Devise. Devise uses this key to generate
+  # random tokens. Changing this key will render invalid all existing
```

Comment:
- [ ] Alongside those features we need to enable some features inside our models/user.rb fille. This features are essential requirements and we need to follow them closely. Please enable the following device modules ```
database_authenticatable, :registerable,:recoverable, :rememberable, :validatable, :confirmable```. we need to make sure `A hashed password is stored in the database.`, `The application asks for confirmation of email.`, `A user can reset the password in the application.`,